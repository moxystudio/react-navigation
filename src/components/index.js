export { default as Navbar } from './navbar';
export { default as Drawer } from './drawer';
