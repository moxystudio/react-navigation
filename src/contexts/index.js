export { default as NavigationContext } from './NavigationContext';
