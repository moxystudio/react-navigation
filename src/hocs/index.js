export { default as withNavigation } from './withNavigation';
