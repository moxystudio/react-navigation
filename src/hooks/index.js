export { default as useDocumentScrollThrottled } from './useDocumentScrollThrottled';
