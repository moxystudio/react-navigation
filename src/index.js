// Components
export { Backdrop, Drawer, Navbar } from './components';

// Providers
export { NavigationProvider } from './providers';

// Hooks
export { useNavigation } from './hooks';

// HOCs
export { withNavigation } from './hocs';
