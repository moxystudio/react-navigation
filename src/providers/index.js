export { default as NavigationProvider } from './NavigationProvider';
