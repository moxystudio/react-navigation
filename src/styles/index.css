@import "./navbar.css";
@import "./drawer.css";
