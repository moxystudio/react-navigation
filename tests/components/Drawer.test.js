import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import { NavigationProvider, Drawer } from '../../src';

const defaultProps = {
    children: <p>Custom Drawer.</p>,
};

const renderWithProps = (props = {}) => render(
    <NavigationProvider>
        <Drawer { ...defaultProps } { ...props } />
    </NavigationProvider>,
);

describe('Drawer Component', () => {
    it('should render correctly', () => {
        const { asFragment } = renderWithProps();

        expect(asFragment()).toMatchSnapshot();
    });

    it('should render correctly when overlay is clicked', () => {
        const { asFragment, container } = renderWithProps();

        const overlay = container.querySelector('.react-navigation_drawer-overlay');

        fireEvent.click(overlay);

        expect(asFragment()).toMatchSnapshot();
    });
});
