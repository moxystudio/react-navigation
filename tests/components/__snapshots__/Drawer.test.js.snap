// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Drawer Component should render correctly 1`] = `
<DocumentFragment>
  <div
    class="react-navigation_drawer-overlay"
    data-open="false"
    data-placement="left"
  />
  <div
    class="react-navigation_drawer react-navigation_drawer-left"
    data-open="false"
    data-placement="left"
  >
    <p>
      Custom Drawer.
    </p>
  </div>
</DocumentFragment>
`;

exports[`Drawer Component should render correctly when overlay is clicked 1`] = `
<DocumentFragment>
  <div
    class="react-navigation_drawer-overlay"
    data-open="false"
    data-placement="left"
  />
  <div
    class="react-navigation_drawer react-navigation_drawer-left"
    data-open="false"
    data-placement="left"
  >
    <p>
      Custom Drawer.
    </p>
  </div>
</DocumentFragment>
`;
